### PR TITLE
fix auto update invalid version detection

### PR DESCRIPTION
**Pull request by ian4hu**
fix auto update invalid version detection.

### CODE REVIEW

**Comment by ian4hu** [comment]
When the github project releases is not enabled, the github release api will response like bellow 

```json
{
  "message": "Not Found",
  "documentation_url": "https://docs.github.com/rest"
}```

then we will get a parsed version `null` which is not empty.

File: scripts/updates/check-updates.sh [hunk 1]
```diff
@@ -49,7 +49,7 @@ if [ -f "${BASEDIR}/github-projects.txt" ]; then
 			latest_version=$(grep -oP "$version_regexp" <<< "$latest_version" || true)
 		fi
 
-		if [ -z "$latest_version" ]; then
+		if [ -z "$latest_version" ] | [ "$latest_version" = "null" ]; then
```
Comment (xtkoba) [inline scripts/updates/check-updates.sh:52]: Did you mean `||`?

Comment (ian4hu) [reply scripts/updates/check-updates.sh:52]: single `|` is expected.

Comment (ian4hu) [reply scripts/updates/check-updates.sh]: Thanks, single `|` is worked but double `||` is more suitable.

Comment (xeffyr) [reply scripts/updates/check-updates.sh]: There is a big difference between `|` and `||`.

**Pull request merged**